kola/tests/docker/docker.go

```diff
@@ -459,17 +470,86 @@ func dockerUserNoCaps(c cluster.TestCluster) {
 	}
 }
 
-// testDockerInfo test that docker info's output is as expected.  the expected
-// filesystem may be asserted as one of 'overlay', 'btrfs', 'devicemapper'
-// depending on how the machine was launched.
-func testDockerInfo(expectedFs string, c cluster.TestCluster) {
+// dockerContainerdRestart ensures containerd will restart if it dies. It tests that containerd is running,
+// kills it, the tests that it came back up.
+func dockerContainerdRestart(c cluster.TestCluster) {
 	m := c.Machines()[0]
 
+	// verify containerd came up
+	if !testContainerdUp(c) {
+		c.Fatalf("Containerd did not start")
+	}
+
+	pid, err := m.SSH("systemctl show containerd -p MainPID")
+	if err != nil {
+		c.Fatal(err)
+	}
+	if string(pid) == "MainPID=0" {
+		c.Fatalf("Could not find containerd pid")
+	}
+
+	// kill it
+	if _, err = m.SSH("sudo kill " + string(pid[8:])); err != nil {
+		c.Fatal(err)
+	}
+
+	// give it a chance to restart
+	if _, err = m.SSH("sleep 2"); err != nil {
```

Comment:
This will cause test flakes on slow nodes.  It'd be better to poll the unit status until it's stable -- either running or stopped.